### PR TITLE
Add IR phase

### PR DESCRIPTION
**Pull request by rynowak**
This change adds a phase which runs IR passes. Design and code are almost
exactly the same as the existing SyntaxTree phase. However all of this is
public because the IR is public API.

### CODE REVIEW

**Comment by rynowak** [comment]
FYI only (unless you're @NTaylorMullen) - going to merge this, but leave open for him in case there is feedback

**Comment by rynowak** [comment]
Ok, well github won't let me. Aw shucks 😭

**Comment by davidfowl** [comment]
I don't know how it all hangs together yet but I approve.

File: src/Microsoft.AspNetCore.Razor.Evolution/IRazorIRPhase.cs [hunk 1]
```diff
@@ -0,0 +1,9 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.AspNetCore.Razor.Evolution
+{
+    public interface IRazorIRPhase : IRazorEnginePhase
```
Comment (NTaylorMullen) [inline src/Microsoft.AspNetCore.Razor.Evolution/IRazorIRPhase.cs]: This isn't used. Also not convinced we need it.

Comment (NTaylorMullen) [inline src/Microsoft.AspNetCore.Razor.Evolution/IRazorIRPhase.cs]: Yup unless that's done in a different PR

**Comment by rynowak** [comment]
OK, I'll remedy that.
